src/caffeine/caffeine_assert/caffeine_assert_s.f90

```diff
@@ -0,0 +1,27 @@
+! Copyright (c), The Regents of the University of California
+! Terms of use are as specified in LICENSE.txt
+submodule(caffeine_assert_m) caffeine_assert_s
```

Comment:
LGTM

Reply:
@bonachea I agreed in our meeting today, but in trying to implement this change, it came up that tests in the test dir use the `assert`, so would lose access to this procedure with this change.

Reply:
@ktras are you then advocating that Caffeine export the assertion library to clients, in addition to implementing the PRIF interface?

I'm not strongly opposed to that idea, but if that's the approach I'd like us to be deliberate and explicit about documenting any "extra" non-PRIF interfaces that the Caffeine implementation is supplying to clients: because if some compiler decides to add a dependency on a non-PRIF Caffeine feature, it could potentially impact our claim that an independent PRIF implementation could be swapped for Caffeine.

Reply:
Alternatively we could create a module of "Caffeine internals" that are used for open-box testing but documented as "not for clients" (and ideally not even installed)

Reply:
As discussed in yesterday's meeting, we are punting the issue of open-box testing and changing the one use of `assert` in the tests to an error stop, since it shouldn't be compiled away anyway.

Reply:
Done.